client/wfprev-war/src/main/angular/src/app/services/util/can-deactive.guard.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { CanDeactivate } from '@angular/router';
+import { Observable } from 'rxjs';
+
+export interface CanComponentDeactivate {
+  canDeactivate: () => Observable<boolean> | Promise<boolean> | boolean;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CanDeactivateGuard implements CanDeactivate<CanComponentDeactivate> {
+  canDeactivate(component: CanComponentDeactivate): Observable<boolean> | Promise<boolean> | boolean {
+    // If the component doesn't have canDeactivate, allow navigation
+    if (!component?.canDeactivate) {
```

Comment:
Would be better to use a hasOwn check here if you're checking for the existence of the attribute. This is checking if component is defined, and then it will check canDeactivate as an attribute (vs function).

`if (!Object.hasOwn(component, 'canDeactivate'))`

You can add a null or undefined check if `component` could be null, but I expect it can't